criu/cr-service.c

```diff
@@ -80,43 +84,47 @@ static int recv_criu_msg(int socket_fd, CriuReq **req)
 		goto err;
 	}
 
-	xfree(buf);
-	return 0;
+	ret = 0;
 err:
-	xfree(buf);
-	return -1;
+	if (buf != (void *)&local)
+		xfree(buf);
+	return ret;
 }
 
 static int send_criu_msg_with_fd(int socket_fd, CriuResp *msg, int fd)
 {
-	unsigned char *buf;
-	int len, ret;
+	u8 local[PB_PKOBJ_LOCAL_SIZE];
+	void *buf = (void *)&local;
+	int len, ret = -1;
 
 	len = criu_resp__get_packed_size(msg);
 
-	buf = xmalloc(len);
-	if (!buf)
-		return -ENOMEM;
+	if (len > sizeof(local)) {
+		buf = xmalloc(len);
+		if (!buf)
+			return -ENOMEM;
+	}
 
 	if (criu_resp__pack(msg, buf) != len) {
 		pr_perror("Failed packing response");
 		goto err;
 	}
 
-	if (fd >= 0) {
+	if (fd >= 0)
 		ret = send_fds(socket_fd, NULL, 0, &fd, 1, buf, len);
-	} else
+	else
 		ret = write(socket_fd, buf, len);
+
 	if (ret < 0) {
 		pr_perror("Can't send response");
 		goto err;
 	}
 
-	xfree(buf);
-	return 0;
+	ret = 0;
 err:
-	xfree(buf);
-	return -1;
+	if (buf != (void *)&local)
+		xfree(buf);
+	return ret;
```

Comment:
pls declare another variable for a function exit code. Usually, we call it exit_code.